compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1826,6 +1832,258 @@ public void visit(BLangWorkerReceive workerReceiveExpr) {
+                ((BLangWorkerMultipleReceive.BLangWorkerReceiveField) receiveField).env
+                        = this.env;
+            } else {
+                dlog.error(
+                        ((BLangWorkerMultipleReceive.BLangWorkerReceiveField) receiveField).workerIdentifier.pos,
+                        DiagnosticCode.INVALID_WORKER_REFERRENCE,
+                        expType.tsymbol
+                          );
+            }
+        }
+
+        if (receiveField.getSendExpression() != null) {
+            BFutureType futureType = new BFutureType(TypeTags.FUTURE, type, null);
+            checkExpr(expression, env, futureType);
+        }
+
+    }
+
+    private BRecordType getWorkerMultipleReceiveReturnType(
+            List<WorkerMultipleReceiveNode.WorkerReceiveFieldNode> receiveFields) {
+
+        BRecordType returnType = new BRecordType(null);
+        for (WorkerMultipleReceiveNode.WorkerReceiveFieldNode receiveField : receiveFields) {
+            if (receiveField.getSendExpression() != null) {
+                BLangIdentifier fieldName;
+                if (receiveField.getWorkerFieldName() != null) {
+                    fieldName = (BLangIdentifier) receiveField.getWorkerFieldName();
+                } else {
+                    fieldName = (BLangIdentifier) receiveField.getWorkerName();
+                }
+
+                BSymbol symbol = symResolver.lookupSymbolInMainSpace(env, names.fromIdNode(fieldName));
+                BType fieldType =
+                        symbol.type.tag == TypeTags.FUTURE ? ((BFutureType) symbol.type).constraint : symbol.type;
+                BField field = new BField(names.fromIdNode(fieldName), null,
+                        new BVarSymbol(0, names.fromIdNode(fieldName), env.enclPkg.packageID,
+                                fieldType, null));
+                returnType.fields.put(field.name.value, field);
+            }
+        }
+
+        returnType.restFieldType = symTable.noType;
+        returnType.sealed = true;
+        returnType.tsymbol = Symbols.createRecordSymbol(0, Names.EMPTY, env.enclPkg.packageID, returnType, null);
+        return returnType;
+    }
+
+    private boolean checkTypeForWorkerMultipleMap(
+            List<WorkerMultipleReceiveNode.WorkerReceiveFieldNode> receiveFields) {
+
+        boolean noSemanticErrors = true;
+        for (WorkerMultipleReceiveNode.WorkerReceiveFieldNode receiveField : receiveFields) {
+            if (checkTypeForWorkerMultipleMap(receiveField, expType)) {
+                noSemanticErrors = false;
+            }
+        }
+        // return boolean depending on whether a semantic error
+        return noSemanticErrors;
+    }
+
+    private boolean checkTypeForWorkerMultipleMap(WorkerMultipleReceiveNode.WorkerReceiveFieldNode receiveField,
+                                                  BType expType) {
+
+        boolean areSemanticErrorsPresent = false;
+
```

Comment:
Shall we extract `((BLangWorkerMultipleReceive.BLangWorkerReceiveField) receiveField)` to a variable and simplify the code. I think we can meet line length requirements also.

Reply:
Fixed by changing the function argument type of `receiveField` to `BLangWorkerMultipleReceive.BLangWorkerReceiveField`

Reply:
nice